### ISSUE TITLE
aws_route53_resolver_rule data lookup tag error

### ISSUE THREAD

**Issue by justicel** [terraform-providers/terraform-provider-aws#10211]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

v0.11.14

### Affected Resource(s)

* data.aws_route53_resolver_rule
* data.aws_route53_resolver_rules

### Terraform Configuration Files

```hcl
data "aws_route53_resolver_rule" "dns_ram" {
  count       = "${length(var.forward_zones)}"
  domain_name = "${var.forward_zones[count.index]}"
  depends_on = ["aws_ram_resource_share_accepter.dns_ram"]
}
```

### Expected Behavior

Should have done a data lookup and located route53 resolver rule(s).

### Actual Behavior

Terraform returns an error that you are not authorized to look at tags on shared resource:

	* module.share.data.aws_route53_resolver_rule.dns_ram[2]: data.aws_route53_resolver_rule.dns_ram.2: error reading Route 53 Resolver rule (rslvr-rr-190b4db83f3b4b148) tags: InvalidParameterException: [RSLVR-00900] Invalid arn: "[RSLVR-00907] You are not authorized to do this operation on shared resource"
	status code: 400, request id: 26e84b24-6b23-48a3-b51f-807ef47ee0cf
	* module.share.data.aws_route53_resolver_rule.dns_ram[4]: data.aws_route53_resolver_rule.dns_ram.4: error reading Route 53 Resolver rule (rslvr-rr-9426511912e444d59) tags: InvalidParameterException: [RSLVR-00900] Invalid arn: "[RSLVR-00907] You are not authorized to do this operation on shared resource"

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

This is a shared resolver rule set from another AWS account. The share was accepted and the rules show up in our account. This appears to be a bug in that you cannot lookup tags of a shared rule via the AWS API.

**Comment by ewbankkit** [comment]
Related:
* https://github.com/terraform-providers/terraform-provider-aws/pull/9085

**Comment by ewbankkit** [comment]
OK, I can reproduce with an acceptance test case:

```console
$ AWS_ALTERNATE_ACCESS_KEY_ID=AAAAAAAAAAAAAAAAAAAA AWS_ALTERNATE_SECRET_ACCESS_KEY=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx make testacc TEST=./aws/ TESTARGS='-run=TestAccDataSourceAwsRoute53ResolverRule_SharedWithMe'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccDataSourceAwsRoute53ResolverRule_SharedWithMe -timeout 120m
=== RUN   TestAccDataSourceAwsRoute53ResolverRule_SharedWithMe
=== PAUSE TestAccDataSourceAwsRoute53ResolverRule_SharedWithMe
=== CONT  TestAccDataSourceAwsRoute53ResolverRule_SharedWithMe
--- FAIL: TestAccDataSourceAwsRoute53ResolverRule_SharedWithMe (215.89s)
    testing.go:569: Step 0 error: errors during apply:
        
        Error: error reading Route 53 Resolver rule (rslvr-rr-5587d23bbd0b4427b) tags: InvalidParameterException: [RSLVR-00907] You are not authorized to do this operation on shared resource
        	status code: 400, request id: e29548bf-4df3-49e5-8539-ce527aea6562
        
          on /var/folders/4j/lf5jdq5j4y91g80ytf6qzjwmk6drp5/T/tf-test261069804/main.tf line 147:
          (source code not available)
        
        
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	215.980s
FAIL
make: *** [testacc] Error 1
```

**Comment by ewbankkit** [comment]
I wonder if the same applies to other RAM [shareable resources](https://docs.aws.amazon.com/ram/latest/userguide/shareable.html) with tags (and data sources)?

* [`aws_ec2_transit_gateway`](https://www.terraform.io/docs/providers/aws/d/ec2_transit_gateway.html)
* [`aws_subnet`](https://www.terraform.io/docs/providers/aws/d/subnet.html)
* [`aws_licensemanager_license_configuration`](https://www.terraform.io/docs/providers/aws/r/licensemanager_license_configuration.html)
* [`aws_rds_cluster`](https://www.terraform.io/docs/providers/aws/d/rds_cluster.html) for Aurora clusters

**Comment by dbaggerman** [comment]
Just checked our TGW shares. `DescribeTransitGateways` in the account being shared to doesn't error, but it returns an empty `Tags` array despite the shared TGW having tags in the source account.

I don't have any shares of the other resource types to check.

**Comment by ewbankkit** [comment]
@dbaggerman Thanks for checking that. It looks like it may just be Aurora RDS clusters that could cause a similar problem; I'll investigate that.
I have finished the PR to fix the Route 53 Resolver rules problem: https://github.com/terraform-providers/terraform-provider-aws/pull/10348.

**Comment by ewbankkit** [comment]
[Aurora cross-account cloning](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.Managing.Clone.html#Aurora.Managing.Clone.Cross-Account) uses the RDS point-in-time restore mechanism.
There is an [PR](https://github.com/terraform-providers/terraform-provider-aws/pull/7031) to support this, I'll add a note to consider investigating this case to that PR.

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.35.0 of the Terraform AWS Provider, Thursday. Thanks to @ewbankkit for the fix. 👍
